audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/SerDesConfig.java

```diff
@@ -132,14 +158,28 @@ private SerDesConfig() {
     /**
      * Kafka value deserialiser for upstream messages coming in as input
      *
+     * @param configProperties contains the directory for error files
      * @return an appropriate value deserialiser for the topic's message content (AuditSuccessMessage)
      */
-    public static Deserializer<AuditSuccessMessage> successValueDeserializer() {
+    public static Deserializer<AuditSuccessMessage> successValueDeserializer(final AuditServiceConfigProperties configProperties) {
         return (String ignored, byte[] auditRequest) -> {
             try {
                 return MAPPER.readValue(auditRequest, AuditSuccessMessage.class);
             } catch (IOException e) {
-                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(auditRequest, Charset.defaultCharset()), e);
+                String failedAuditString = new String(auditRequest, Charset.defaultCharset());
+                String fileName = "Success-" + ZonedDateTime.now(ZoneOffset.UTC)
+                        .format(DateTimeFormatter.ISO_INSTANT).replace(":", "-");
+                File directory = new File(configProperties.getErrorDirectory());
+                File parent = directory.getAbsoluteFile().getParentFile();
+                File timestampedFile = new File(parent, fileName);
+                try (FileWriter fileWriter = new FileWriter(timestampedFile, Charset.defaultCharset(), !timestampedFile.createNewFile())) {
+                    fileWriter.write(failedAuditString);
+                    LOGGER.info("Failed to deserialize a 'success' audit message. Created file {}", timestampedFile);
+                } catch (IOException ex) {
+                    LOGGER.error("Failed to process audit request '{}'", failedAuditString, ex);
+                }
+                SerDesHealthIndicator.addSerDesExceptions(e);
+                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + failedAuditString, e);
```

Comment:
Suggest pull this out into a method to prevent the code duplication

Reply:
Extracted this code into a separate method.